File: content-solutions/two-pointers/good-weeks.py
def solution(scores, k, threshold):
    need = threshold * k
    window = sum(scores[:k])
    count = 1 if window >= need else 0
    for i in range(k, len(scores)):
        window += scores[i] - scores[i - k]
        if window >= need:
            count += 1
    return count

File: content-solutions/two-pointers/good-weeks.java
class Solution {
    public int solution(int[] scores, int k, int threshold) {
        long need = (long) threshold * k;
        long window = 0;
        for (int i = 0; i < k; i++) window += scores[i];
        int count = window >= need ? 1 : 0;
        for (int i = k; i < scores.length; i++) {
            window += scores[i] - scores[i - k];
            if (window >= need) count++;
        }
        return count;
    }
}
